sunpy/net/base_client.py

```diff
@@ -102,7 +104,70 @@ def show(self, *cols):
         table = self.build_table()
         if len(cols) == 0:
             return table
-        return table[list(cols)]
+        tablecols = table.columns
+        valid_cols = [col for col in cols if col in tablecols]
+        return table[valid_cols]
+
+
+class BaseQueryResponseTable(BaseQueryResponse):
+    """
+    A base class for tabular results returned from clients.
+    """
+
+    def __init__(self, table=None, client=None):
+        super().__init__()
+        self.table = table or QTable()
+        self._client = client
+
+    @property
+    def client(self):
+        return self._client
+
+    @client.setter
+    def client(self, client):
+        self._client = client
+
+    @property
+    def blocks(self):
+        return list(self.table.iterrows)
+
+    def __len__(self):
+        if self.table:
+            return len(self.table)
+        else:
+            return 0
+
+    def __getitem__(self, item):
+        if isinstance(item, int):
+            item = slice(item, item + 1)
+        ret = type(self)(self.table[item])
+        ret.client = self._client
+        return ret
+
+    def __iter__(self):
+        return (t for t in [self])
```

Comment:
I think this can be:
```suggestion
        yield self
```
thought it should probably have a test to make sure.

Reply:
That being said, is this really the behaviour we want here? Why not iterate over the rows? Or not iterate at all?

Reply:
Fixed?!